pkg/cmd/cli/describe/chaindescriber.go

```diff
@@ -0,0 +1,267 @@
+package describe
+
+import (
+	"fmt"
+	"strings"
+
+	utilerrors "github.com/GoogleCloudPlatform/kubernetes/pkg/util/errors"
+	dot "github.com/awalterschulze/gographviz"
+	"github.com/golang/glog"
+	"github.com/gonum/graph"
+
+	osgraph "github.com/openshift/origin/pkg/api/graph"
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	buildedges "github.com/openshift/origin/pkg/build/graph"
+	buildutil "github.com/openshift/origin/pkg/build/util"
+	"github.com/openshift/origin/pkg/client"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	imageedges "github.com/openshift/origin/pkg/image/graph"
+	imagegraph "github.com/openshift/origin/pkg/image/graph/nodes"
+	"github.com/openshift/origin/pkg/util/parallel"
+)
+
+// ChainDescriber generates extended information about a chain of
+// dependencies of an image stream
+type ChainDescriber struct {
+	client.Interface
+	Output string
+}
+
+// NewChainDescriber returns a new ChainDescriber
+func NewChainDescriber(c client.Interface, out string) *ChainDescriber {
+	return &ChainDescriber{Interface: c, Output: out}
+}
+
+// MakeGraph will create the graph of all build configurations and the image streams
+// they point to via image change triggers in the provided namespace
+func (d *ChainDescriber) MakeGraph(namespace string) (osgraph.Graph, error) {
+	g := osgraph.New()
+
+	loaders := []GraphLoader{
+		&bcLoader{namespace: namespace, lister: d.Interface},
+		&isFromBuildConfigLoader{namespace: namespace, lister: d.Interface, bc: &bcLoader{namespace: namespace, lister: d.Interface}},
+	}
+	loadingFuncs := []func() error{}
+	for _, loader := range loaders {
+		loadingFuncs = append(loadingFuncs, loader.Load)
+	}
+
+	if errs := parallel.Run(loadingFuncs...); len(errs) > 0 {
+		return g, utilerrors.NewAggregate(errs)
+	}
+
+	for _, loader := range loaders {
+		loader.AddToGraph(g)
+	}
+
+	buildedges.AddAllInputOutputEdges(g)
+	imageedges.AddAllImageStreamRefEdges(g)
+
+	return g, nil
+}
+
+func (d *ChainDescriber) Describe(namespace, input string) (string, error) {
+	g, err := d.MakeGraph(namespace)
+	if err != nil {
+		return "", err
+	}
+	name, tag, all, err := parseInput(input)
+	if err != nil {
+		return "", err
+	}
+	if all {
+		// TODO: Dump all the graph
+		return "", nil
+	}
+
+	// Find out if the provided stream exists
+	exists := false
+	var stream graph.Node
+	for _, node := range g.NodesByKind(imagegraph.ImageStreamNodeKind) {
+		if g.GraphDescriber.Name(node) == name {
+			stream = node
+			exists = true
+			break
+		}
+	}
+	if !exists {
+		return "", fmt.Errorf("image stream %q does not exist", name)
+	}
+
+	if d.Output == "dot" {
+		return dotDump(g, stream, tag)
+	}
+	return g.String(), nil
```

Comment:
I wasn't intending to use it. Though we may have a problem here. Previously build-chain was using json, dot, and abstract syntax tree format. Now, how can I produce a json graph since we don't map json fields in our structs? Can we do that? See [here](https://github.com/openshift/origin/blob/9a3b2ba8fb5547c8812e376e166b278d645b40a9/pkg/cmd/experimental/buildchain/buildchain.go#L51-L54). We are generating a graph of those structs, then pass the root to the [json marshaler](https://github.com/openshift/origin/blob/9a3b2ba8fb5547c8812e376e166b278d645b40a9/pkg/cmd/experimental/buildchain/buildchain.go#L217-L222) and the json graph is automatically generated.